src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -281,21 +281,24 @@ namespace OpenMS
         AASequence seq = h.getSequence();
         double precursor_neutral_mass = seq.getMonoWeight();
 
-        Int scan_index;
+        int scan_index = count;
+        int scan_nr = 0;
+        
         if (lookup.empty())
         {
           if (it->metaValueExists("RT_index")) // Setting metaValue "RT_index" in XTandemXMLFile in the case of X! Tandem.
           {
-            scan_index = it->getMetaValue("RT_index");
-          }
-          else
-          {
-            scan_index = count;
+            scan_index = it->getMetaValue("RT_index");            
           }
+          scan_nr = scan_index + 1;          
```

Comment:
As far as I know is in most cases between the scan_index and the scan_nr a difference of 1. I tried to reproduce this pattern with scan_index + 1. But yes, this might be not necessary and if no spectrum lookup is possible, it should be enough to set scan_nr to scan_index to keep it simple.